pyprof/prof/normalization.py

```diff
@@ -28,49 +26,42 @@ def __init__(self, d):
         op = marker['op']
         args = marker['args']
 
-        self.marker = marker
-        self.mod_ = mod
-        self.op_ = op
-        self.args = args
+        self._mod = mod
+        self._op = op
 
         assert (op == "batch_norm")
         assert (len(args) >= 1)
         i = args[0]
         assert (i['type'] == "tensor")
 
-        self.shape = i['shape']
-        self.type = i['dtype']
+        self.inp = Tensor(i['shape'], i['dtype'])
         self.dir = d.dir
         self.sub = d.sub
 
     def params(self):
-        p = OrderedDict([('T', self.shape), ('type', self.type)])
-        return p
+        return str(self.inp)
 
     def tc(self):
         return "-"
 
     def op(self):
-        return self.op_
+        return self._op
 
     def mod(self):
-        return self.mod_
-
-    def elems(self):
-        return Utility.numElems(self.shape)
+        return self._mod
 
     def flops(self):
         # Variance algo-dependent, but this is a reasonable value.
-        return self.elems() * 8
+        return self.inp.size * 8
```

Comment:
Why 8? Could this more controllable? Another named constant would help, i.e.

```
return self.inp.size * FLOP_MULTIPLY_FACTOR
```

In the future, do you think that this multiply factor may be fine tuned for the architecture or algorithm?

Reply:
The multiplication factor (8 in this case) depends on the operation (batch normalization in this case) and the direction in which it is called (forward or backward), so the constant is very specific to this op and would not be reused anywhere else (so hoisting it out would not be useful).